backend/src/data/repositories/user-details/user-details.repository.ts

```diff
@@ -0,0 +1,45 @@
+import { UserDetailsUpdateInfoRequestDto } from '~/common/types/types';
+import { UserDetails as UserDetailsM } from '~/data/models/models';
+
+type Constructor = {
+  UserDetailsModel: typeof UserDetailsM;
+};
+
+class UserDetails {
+  #UserDetailsModel: typeof UserDetailsM;
+
+  public constructor({ UserDetailsModel }: Constructor) {
+    this.#UserDetailsModel = UserDetailsModel;
+  }
+
```

Comment:
i think we also need a `create` method. since we need to create a new record on the user registration

Reply:
@what1s1ove 
Maybe it could be renamed to `upsertUserDetails` to avoid misunderstanding